src/common/Throttle.cc

```diff
@@ -92,8 +92,10 @@ bool Throttle::_wait(int64_t c, UNIQUE_LOCK_T(lock)& l)
   if (_should_wait(c) || !conds.empty()) { // always wait behind other waiters.
     {
       auto cv = conds.emplace(conds.end());
+      conds_empty = conds.empty();
       auto w = make_scope_guard([this, cv]() {
 	  conds.erase(cv);
+	  conds_empty = conds.empty();
 	});
       waited = true;
```

Comment:
I'm worried about a race between someone here and another thread calling put().  I'm having trouble convincing myself it's not possible for a waiter to wait when the put() dropped it back below the max+c.

Reply:
I think we need to recheck _should_wait() here